include/wsrep/xid.hpp

```diff
@@ -21,23 +21,80 @@
 #define WSREP_XID_HPP
 
 #include <iosfwd>
-#include <string>
+#include "buffer.hpp"
 
 namespace wsrep
 {
     class xid
     {
     public:
-        xid() : xid_() { }
-        xid(const std::string& str) : xid_(str) { }
-        xid(const char* s) : xid_(s) { }
-        bool empty() const { return xid_.empty(); }
-        void clear() { xid_.clear(); }
-        bool operator==(const xid& other) const { return xid_ == other.xid_; }
+        xid()
+            : format_id_(-1)
+            , gtrid_len_(0)
+            , bqual_len_(0)
+            , data_()
+        { }
+
+        xid(long format_id, long gtrid_len,
```

Comment:
Maximum values for `gtrid_len` and `bqual_len` are 64. Maybe this should be at least asserted in this constructor.